src/app/students/(hooks)/useStudents.tsx

```diff
@@ -12,54 +21,111 @@ type StudentsContextType = {
+// todo
+const FIXED_COURSE_NUM = 1
 
 const useStudentsProvider = (): StudentsContextType => {
+    const pathname = usePathname()
+    const searchParams = useSearchParams()
+
     const [displayStudents, setDisplayStudents] = useState<Student[]>([])
     const [sections, setSections] = useState<DropdownOption[]>([])
-    const [totalStudents, setTotalStudents] = useState<number>(0)
-    const searchParams = useSearchParams()
+    const [totalStudents, setTotalStudents] = useState(0)
+    const [search, setSearch] = useState(searchParams.get('search') ?? '')
+    const [sectionTerms, setSectionTerms] = useState(searchParams.get('sections') ?? '')
+    const [pageCount, setPageCount] = useState(0)
+    const [pagination, setPagination] = useState<PaginationState>({
+        pageIndex: parseInt(searchParams.get('page') ?? '1') - 1,
+        pageSize: parseInt(searchParams.get('per_page') ?? '10'),
+    })
 
+    const setSearchQuery = useCallback((searchQuery: string) => {
+        setSearch(searchQuery)
+        setPagination({ ...pagination, pageIndex: 0})
+    }, [pagination, setSearch, setPagination])
 
-    const fixedCourseNum = 1
+    const filterSections = useCallback((sectionsPassed: DropdownOption[]) => {
+        setSectionTerms(sectionsPassed.map((section) => section.value).join(','))
+        setPagination({ ...pagination, pageIndex: 0})
+    }, [setSectionTerms, setPagination, pagination])
 
-    const { pageIndex, pageSize } = useMemo(() => {
-        return {
-            pageIndex: parseInt(searchParams.get('page') ?? '1'),
-            pageSize: parseInt(searchParams.get('per_page') ?? '10'),
-        }
-    }, [searchParams])
+    const queryStringParams: QueryParams = useMemo(() => ({
+        page: pagination.pageIndex + 1, // adding one to make it appear as 1 indexed for the URL query in browser
+        per_page: pagination.pageSize,
+        search: search,
+        sections: sectionTerms.length > 0 ? sectionTerms : undefined,
+    }), [pagination.pageIndex, pagination.pageSize, search, sectionTerms])
 
     useEffect(() => {
+        const queryString = createQueryString(queryStringParams)
         const fetchStudents = async () => {
-            const baseURL = process.env.NEXT_PUBLIC_BACKEND_URL
-            const courseMemberData = await fetch(`${baseURL}/api/v1/course-members/course/${fixedCourseNum}/?page=${pageIndex}&per_page=${pageSize}`).then(res => res.json())
+            const courseMemberResponse = await fetch(`${process.env.NEXT_PUBLIC_BACKEND_URL}/api/v1/course-members/course/${FIXED_COURSE_NUM}/?${queryString}`)
+            const courseMemberData = await courseMemberResponse.json()
             const studentsToDisplay: Student[] = courseMemberData.results.map((member: any) => ({
-                id: member.id,
-                name: member.user.last_name + ',' + member.user.first_name,
+                id: member.user.id,
+                name: `${member.user.last_name}, ${member.user.first_name}`,
```

Comment:
nit; (not for this PR)

now that I think about it, probably unwise to concatenate this here, only to display them in separate columns anyway yknow

Reply:
will make an issue for it